apps/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,42 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 
 @Component({
   selector: 'app-card',
-  templateUrl: './card.component.html',
-  standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
-})
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
-
-  CardType = CardType;
+  template: `
+    <ng-content select="[card-header]" />
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
+    <section>
+      <ng-container *ngFor="let item of list">
+        <ng-template
+          *ngTemplateOutlet="row ?? empty; context: { $implicit: item }" />
+        <ng-template #empty>
+          <div>No template</div>
+        </ng-template>
+      </ng-container>
+    </section>
 
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+    <button
+      class="border border-blue-500 bg-blue-300 p-2 rounded-sm"
+      (click)="added.emit()">
+      Add
+    </button>
+  `,
+  standalone: true,
+  host: {
+    class: 'border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3',
+  },
+  imports: [NgIf, NgFor, NgTemplateOutlet],
+})
+export class CardComponent<T> {
+  @Input({ required: true }) list: T[] = [];
+  @ContentChild('row', { read: TemplateRef }) row: TemplateRef<any>;
```

Comment:
Minor: to avoid magic string, you can use a directive instead of `row`

Minor: You can also type your template, instead of using any.

Reply:
It's fixed, thanks for your feedback 😁